xarray/core/dataset.py

```diff
@@ -1449,6 +1449,13 @@ def __len__(self) -> int:
         return len(self.data_vars)
 
     def __bool__(self) -> bool:
+        warnings.warn(
+            "coercing a Dataset to a bool be deprecated in the future. Using "
+            "len(ds) > 0 to check for data_variables or converting to an array with "
```

Comment:
Shall we  recommend `bool(ds.data_vars)` instead?

Reply:
sure! that makes sense :)